Controllers/LeaderboardController.cs

```diff
@@ -0,0 +1,37 @@
+
+using Microsoft.AspNetCore.Mvc;
+
+
+namespace PlayersList;
+
+public class LeaderboardController : Controller
+{
+
+    public class Player
+    {
+        public int PlayerID { get; set; }
+        public string? PlayerFirstName { get; set; }
+        public int Age { get; set; }
+    
+
+        IList<Player> playersList = new List<Player>() {
+
+                new Player(){ PlayerID=1, PlayerFirstName="Steve", Age = 21 },
+                new Player(){ PlayerID=2, PlayerFirstName="Bill", Age = 25 },
+                new Player(){ PlayerID=3, PlayerFirstName="Ram", Age = 20 },
+                new Player(){ PlayerID=4, PlayerFirstName="Ron", Age = 31 },
+                new Player(){ PlayerID=5, PlayerFirstName="Rob", Age = 19 }
+            };
+    }
+    
+
+    public ActionResult Leaderboard()
+
+   {
+            ViewBag.Message = "Hello";
```

Comment:
ooh - so easy mistake to make - you've added this in the `LeaderboardController` as a `Leaderboard` action - so using the MVC routing which goes `Controller/Action` - this will get called if you go to the URL `Leaderboard/Leaderboard`

You also have a Leaderboard action in your Home Controller, which is how you're currently routing traffic at `Home/Leaderboard`.

I'd get rid of one of the leaderboards to help prevent the confusion in the future.  If it has it's own controller, then you can usually name the default route to render as `Index` and that way you could route their using just `/Leaderboard` as the url.  If you're using the default action, you don't need to include in the URL.  Then also make sure you have the `Views` folder setup appropriately too.  When you just return `View()` without providing an _explicit_ path, then the MVC conventions take over and it'll be looking for a folder at `Views/Leaderboard/`